### ISSUE TITLE
Add animation parameter to AnimationNodeAnimation

### ISSUE THREAD

**Issue by ODtian** [godotengine/godot-proposals#8818]
### Describe the project you are working on

A animation controller plays many custom animation.

### Describe the problem or limitation you are having in your project

Currently animation property is implemented as a property in a resource class `AnimationNodeAnimation` which is bind with a resource, if I want to change it at runtime     , It changes animation for all characters referring this resource.

Since I have many animation, and is ensured until runtime, enumerating all of them is not working.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Fortunately, ´AnimationTree´ provide parameters, which is properties which bind to AnimationTree node, change it at runtime will only affect that instance.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add animation parameter in ´AnimationNodeAnimation´ 's ´get_parameters_list´ method, then use the parameter value anywhere in class ´AnimationNodeAnimation´ if it was set, otherwise use resource's property as default value.

In short, let parameter override the property.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

This improvement require changing in ´AnimationNode´ 's ´_process´ method, however it which is hard to rewrite.

**Comment by TokageItLab** [comment]
As I mentioned in https://github.com/godotengine/godot/pull/87111#issuecomment-1889678948, I don't see any valid reason why this is necessary.

You may want to change the animation more easily than a NodeTransition or StateMachine, but this is more of a risk.

**Comment by sunbromarko** [comment]
@TokageItLab I wanna swap animation set when my char equip weapon, i have 18 animation sets now stored in Resources, and custom script swapping animations in nodes. With "common way" every new set force me edit tree instead just create Resource.

**Comment by sunbromarko** [comment]
Just imagine game like Dark souls with "Gesture". You can have OneShot Node with 100500 animations variants for player to choose. https://www.youtube.com/watch?v=hK9aiBJlfN8
How should i do it without override function ?

**Comment by TokageItLab** [comment]
As mentioned above, you can create a helper script that places all animations within StateMachine's Nested mode.

Even if we were to incorporate equivalent functionality, I believe it would need to be a node other than NodeAnimation at the very least. As for the behavior, something like a NodeAnimationList that internally holds transitions for multiple registered animations could be considered.

**Comment by sunbromarko** [comment]
I'm working on recreating the character controller from Gothic 1. Let me give an example using the melee combat mechanics.

At the base there aren't that many states: strike right, strike left, and a combo chain (up to 4 attacks). 

<img width="565" height="499" alt="Image" src="https://github.com/user-attachments/assets/0ca42d88-ccd6-468f-ab42-653393e577a6" />

The complexity comes later, though. There are 2 weapon types — one-handed and two-handed — each with its own animations. On top of that, the animation set depends on the character's weapon skill, with 3 skill levels per weapon.

To me it feels completely natural to want to simply swap the correct animations into these Nodes at the moment the player decides to draw their weapon. Right now I'm using a "hack" and swapping these animations in the nodes via a script like this, but I dream of the day when I'll be able to simply plug a parameter into the blackboard. I just want to keep my tree as clean as possible.

<img width="675" height="496" alt="Image" src="https://github.com/user-attachments/assets/f9479a58-230c-47ff-944b-90060a7cc822" />

I have to do the same kind of hacks for one-shot gestures, one-shot special animations in dialogues, equip/unequip weapon animations, several dozen spells, and so on.
